kubernetes/kops
# Release 1.24.2

- Add hashes for latest Docker versions
- Update Docker to v20.10.17
- Run hack/update-expected.sh
- Add hashes for latest containerd versions
- Update containerd to v1.6.6
- Run hack/update-expected.sh
- Update containerd fallback to v1.4.13
- Channels to have exit status 1 on apply failure
- Fix codegen targets and whitespace errors in Makefile
- Add support for setting mode field on file assets
- Update documentation for fileAssets and fix whitespace error
- Revert "Use kubectl replace instead of apply when updating addons"
- Fix API group being incorrect for ingresses
- Update after running hack/update-expected.sh
- Update runc to v1.1.3
- Run hack/update-expected.sh
- Update AWS CCM images for k8s 1.20-1.22
- Run hack/update-expected.sh
- Fix namespace for cert manager webhook config
- Avoid spurious changes with ed25519 keys
- Add back the metrics-server 443 port with a new name
- Fix broken node selector for node termination handler
- Release 1.24.0-beta.2 (#13788)
- replace flexdriver with busybox
- update expected
- Update etcd-manager to v3.0.20220617
- Run hack/update-expected.sh
- Fix tests
- Do not run CAS on spot instances
- Fix GCE resource tracking
- Limit GCE ASG labels to 63 chars
- Run hack/update-expected.sh
- Adding GuestAccelerators to InstanceTemplate
- Limit GCE tag for role to 63 chars
- Replace manifests after apply
- Fix upgrade-ab skip e2e test
- Don't try to manage the kube-system namespace
- Run hack/update-expected.sh
- Remove unneeded kube-proxy service account
- Move kube-dns service account to kube-dns addon
- Completely remove core addons
- Run hack/update-expected.sh
- Disable removal or CCM leader migration
- Release 1.24.0-beta.3
- Clean-up firewall rules that contain targets with the cluster name hash
- Add integration test for GCE cluster with very long cluster name
- Log errors from detachInstance
- gce: Move out of beta, drop feature flag
- Run make gen-cli-docs
- gce: set ProvisioningModel on InstanceTemplate
- Fix cleanup of firewall rules that contain the cluster name hash
- Apply PKI even if addon fails
- Update dependencies
- Refactor ClusterPrefixedName and ClusterSuffixedName to not return error
- Mount /etc/hosts from host for CoreDNS
- Run hack/update-expected.sh
- Limit GCE names to 63 chars for various resources
- Make IRSA webhook configure apps to use regional STS and set the default region on them
- Make it possible to enable the shield addon for LBC
- Increase length of cluster name for GCE long cluster name integration test
- Add integration test for GCE cluster with internal LB and very long cluster name
- Run hack/update-expected.sh
- Limit GCE router name to 63 chars
- Run hack/update-expected.sh
- Remove the v1alpha3 API version
- Update Cilium to 1.11.6
- Fix unsetting ASG max price
- Revert "Add back the metrics-server 443 port with a new name"
- aws: introduce maximum instance lifetime in cluster
- Fix doc of NewOpenStackCloudProvider
- Add config drive as a source for OpenStack instance metadata
- Be more specific when filtering OS instance ports
- Use csi-snapshotter for OS only when the controller is enabled
- Bump EBS CSI driver to 1.8.0
- Run hack/update-expected.sh
- Release 1.24.0 (#13957)
- Use Calico v3.23 for Kubernetes 1.22+
- Run hack/update-expected.sh
- Use control-plane node role for AWS IAM Authenticator
- Enable AWS IAM Authenticator in complex integration test
- Upgrade complex integration test case to k8s 1.24
- Skip deregistering the instance during rolling update for Spotinst
- Upgrade aws-iam-authenticator to v0.5.9
- Add option to set etcd-manager backup interval
- Use only IPv4 for Hetzner servers
- Add option to set number of replicas for pod-identity-webhook
- Update etcd-manager to v3.0.20220717
- Run hack/update-expected.sh
- Update Go to v1.18.4
- Remove replaces from go.mod
- Update k8s.io/client-go to match k8s.io/api
- Run "make gomod"
- Upgrade DO CSI driver to 4.2.0
- Update Calico to v3.23.2
- Update Calico to v3.23.3
- Update Canal to v3.23.3
- Run hack/update-expected.sh
- Switch to latest MacOS version for CI
- Update dependencies
- Revert to using instance private DNS name to lookup hostname
- Add server group management for Hetzner
- Update etcd-manager to v3.0.20220727
- Run hack/update-expected.sh
- Check keyset existence before attempting to distrust
- Fix SIGSEGV when deleting a Hetzner instance
- Remove namespaces from cluster-scoped resources in CNI manifests
- ./hack/update-expected.sh
- Enable rolling updates for Hetzner
- Wait for load balancer to be ready for Hetzner
- Add multiple SSH keys support for Hetzner
- Use cabundle for etcd CA files
- Release 1.24.1 (#14071)
- Allow configuring OpenStack CCM networking options
- aws-ebs-csi-driver: remove preStop hook
- fixup! aws-ebs-csi-driver: remove preStop hook
- Add option to configure runc version for containerd
- Run hack/update-expected.sh
- Bump nvidia device plugin to 0.12.0
- Add hashes for containerd v1.6.7
- Update containerd to v1.6.8
- Run hack/update-expected.sh
- Disable some flags in kube-controller-manager and kube-scheduler when logging-format is not text
- Add deployment-specific selectors to nth pdb
- Bump the CCM images
- Limit GCE network names to 63 chars
- Run hack/update-expected.sh
- Update runc to v1.1.4
- Run hack/update-expected.sh
- OIDC: Tolerate extra service-account key set items
- Always disable rp_filter when using cilium
- Bump cilium to 1.11.8
- Bump cert-manager to 1.8.2
- Release 1.24.2
